src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -146,17 +135,17 @@ SIValue AR_TOLOWER(SIValue *argv, int argc) {
 	char *original = argv[0].stringval;
 	short lower_len = strlen(original);
 	char *lower = rm_malloc((lower_len + 1) * sizeof(char));
-	_toLower(original, lower, lower_len);
+    str_tolower(original, lower, &lower_len);
 	return SI_TransferStringVal(lower);
 }
 
 /* returns the original string in uppercase. */
 SIValue AR_TOUPPER(SIValue *argv, int argc) {
 	if(SIValue_IsNull(argv[0])) return SI_NullVal();
 	char *original = argv[0].stringval;
-	size_t upper_len = strlen(original);
+	short upper_len = strlen(original);
```

Comment:
Why the change from `size_t` to `short` ?

Reply:
because I saw that "short" is used in every other place for storing the string length. Is it matter?

Reply:
Yeah sort and size_t are probably of different sizes, also `strlen` returns `size_t`